src/NHibernate/ISession.cs

```diff
@@ -764,6 +764,11 @@ public partial interface ISession : IDisposable
 		/// </summary>
 		ITransaction Transaction { get; }
 
+		/// <summary>
+		/// The current transaction if any, else <see langword="null" />.
+		/// </summary>
+		ITransaction CurrentTransaction { get; }
```

Comment:
With 2 commits breaking changes are spotted quite easily :) Do we really need it?

Reply:
Indeed I forgot, have to add it as an extension method, ...

We need it for obsoleting `ISession.Transaction`, unless we consider the return value from `BeginTransaction` is enough and that is the user concern to pass around the transaction if he needs it / needs to check if a transaction is ongoing.
Otherwise we could keep that `Transaction` property non-obsoleted, and just do a breaking change at 6.0. But it is not developer friendly to do breaking changes without warning in previous versions allowing to anticipate them.